File: robert-server-batch/src/test/java/fr/gouv/stopc/robertserver/batch/test/CountryCode.java
package fr.gouv.stopc.robertserver.batch.test;

import com.google.protobuf.ByteString;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.math.BigInteger;

@RequiredArgsConstructor
public enum CountryCode {

    GERMANY(49),
    FRANCE(33);

    @Getter
    private final int numericCode;

    public byte[] asByteArray() {
        return BigInteger.valueOf(numericCode).toByteArray();
    }

    public ByteString asByteString() {
        return ByteString.copyFrom(asByteArray());
    }

    @Override
    public String toString() {
        return String.format("%s(%d)", name(), numericCode);
    }
}

File: robert-crypto-grpc-server/src/test/kotlin/fr/gouv/stopc/robertserver/crypto/test/CountryCode.kt
package fr.gouv.stopc.robertserver.crypto.test

import com.google.protobuf.ByteString

enum class CountryCode(val numericCode: Int) {

    GERMANY(49),
    FRANCE(33);

    fun asByteArray() = arrayOf(numericCode.toByte()).toByteArray()

    fun asByteString(): ByteString = ByteString.copyFrom(asByteArray())

    override fun toString() = "$name($numericCode)"
}
